src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -2262,16 +2262,23 @@ public bool HasValue
 
         internal AttributeInfo FindTargetAttribute(EntityHandle hasAttribute, AttributeDescription description)
         {
-            return FindTargetAttribute(MetadataReader, hasAttribute, description);
+            return FindTargetAttribute(MetadataReader, hasAttribute, description, out _);
         }
 
-        internal static AttributeInfo FindTargetAttribute(MetadataReader metadataReader, EntityHandle hasAttribute, AttributeDescription description)
+        internal static AttributeInfo FindTargetAttribute(MetadataReader metadataReader, EntityHandle hasAttribute, AttributeDescription description, out bool foundAttributeType)
         {
+            foundAttributeType = false;
+
             try
             {
                 foreach (var attributeHandle in metadataReader.GetCustomAttributes(hasAttribute))
                 {
-                    int signatureIndex = GetTargetAttributeSignatureIndex(metadataReader, attributeHandle, description);
+                    bool matchedAttributeType;
+                    int signatureIndex = GetTargetAttributeSignatureIndex(metadataReader, attributeHandle, description, out matchedAttributeType);
+                    if (matchedAttributeType)
+                    {
+                        foundAttributeType = true;
```

Comment:
Shouldn't this be inside the `if (signatureIndex != -1)` block? Consider that as written the following behavior happens:

1. There are two attributes
2. The first one matches the type but has the wrong signature
3. The second one throws an exception 

The method will end up returning `default` but has `foundAttributeType == true`. Is that the intended behavior?

Reply:
> The method will end up returning `default` but has `foundAttributeType == true`. Is that the intended behavior?

Yes, that was the intended behavior. `foundAttributeType` is useful specifically for cases where there is no signature match.